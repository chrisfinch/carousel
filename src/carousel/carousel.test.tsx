import React from "react";
import ReactDOMServer from "react-dom/server";
import Carousel from "./carousel.component";
import PixaBayFixture from "./data/pixabay.json";
import { fromJS } from "immutable";
import { PixaBayData } from "../types";

jest.mock("./components/image-when-loaded.component", () => () => (
	<div>image mock</div>
));

const immutablePixaBayFixture = fromJS(
	PixaBayFixture
) as PixaBayData.IImmutableResponse;

describe("Carousel", () => {
	it("renders correctly", () => {
		const wrapper = ReactDOMServer.renderToString(
			<Carousel
				images={immutablePixaBayFixture.get("hits")}
				title="Carousel Test"
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
