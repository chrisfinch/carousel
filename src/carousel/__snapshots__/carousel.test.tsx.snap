// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `"<style data-emotion-css=\\"79elbk\\">.css-79elbk{position:relative;}</style><div class=\\"carousel css-79elbk\\"><style data-emotion-css=\\"e94x7s\\">.css-e94x7s{background-color:#E4EBE0;padding:16px;}@media (max-width:375px){.css-e94x7s{padding-bottom:32px;}}</style><div class=\\"carousel__title css-e94x7s\\"><style data-emotion-css=\\"1eeg7ul\\">.css-1eeg7ul{font-family:\\"Ropa Sans\\",sans-serif;color:#035288;font-size:42px;}</style><div class=\\"css-1eeg7ul\\">Carousel Test</div></div><style data-emotion-css=\\"16xm07y\\">.css-16xm07y{padding:16px 14px;}</style><div class=\\"carousel__contents css-16xm07y\\"><style data-emotion-css=\\"19gghux\\">.css-19gghux{width:100%;overflow:hidden;position:relative;}</style><div class=\\"carousel__container css-19gghux\\"><style data-emotion-css=\\"1kozlqq\\">.css-1kozlqq{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;list-style:none;margin:0;padding:0;overflow:hidden;position:relative;left:0;-webkit-transition:left 0.2s ease-in-out;transition:left 0.2s ease-in-out;}</style><ul class=\\"carousel__images css-1kozlqq\\"><style data-emotion-css=\\"1y2v8sk\\">.css-1y2v8sk{width:200px;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;padding:0;position:relative;box-sizing:border-box;margin-right:8px;}.css-1y2v8sk:last-child{margin-right:0;}</style><li class=\\"carousel__image-container css-1y2v8sk\\"><style data-emotion-css=\\"b6jvmt\\">.css-b6jvmt{overflow:hidden;position:relative;width:100%;height:200px;}</style><div class=\\"image__frame css-b6jvmt\\"><style data-emotion-css=\\"y9uc39\\">.css-y9uc39{position:absolute;left:50%;height:100%;}</style><div class=\\"image__center css-y9uc39\\"><style data-emotion-css=\\"gxts0j\\">.css-gxts0j{display:block;height:100%;position:relative;left:-50%;}</style><div>image mock</div></div></div><style data-emotion-css=\\"b8kzem\\">.css-b8kzem{padding-top:12px;}</style><div class=\\"image__meta css-b8kzem\\"><style data-emotion-css=\\"ozpmrz\\">.css-ozpmrz{font-family:\\"Ropa Sans\\",sans-serif;color:#035288;font-size:16px;margin-bottom:6px;text-overflow:ellipsis;overflow:hidden;white-space:nowrap;}@media (max-width:375px){.css-ozpmrz{font-size:24px;}}</style><div class=\\"image__title css-ozpmrz\\">fantasy, beautiful, dawn</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->peter_pyw</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">green, park, season</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->valiunic</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">barley field, wheat, harvest</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->kangbch</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">italy, sunrise, sky</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">castelmezzano, italy, village</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">angel, nature, clouds</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->Comfreak</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">utah, mountain, sky</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->Fotocitizen</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">forest, path, mystical</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->SplitShire</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">mountains, sunrise, dawn</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">polynesia, french polynesia, tahiti</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->Julius_Silver</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">lake irene, colorado, water</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">sunrise, boat, rowing boat</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->quangle</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">godafoss, iceland, waterfall</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">rainbow, roadway, beautiful</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->Free-Photos</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">winter, snow, sunrise</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">river, hills, evening</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">forest, trees, woods</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">sunset, dawn, nature</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->kareni</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">alberta, canada, lake</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->jameswheeler</div></div></li><li class=\\"carousel__image-container css-1y2v8sk\\"><div class=\\"image__frame css-b6jvmt\\"><div class=\\"image__center css-y9uc39\\"><div>image mock</div></div></div><div class=\\"image__meta css-b8kzem\\"><div class=\\"image__title css-ozpmrz\\">iceland, aurora borealis, northern lights</div><div class=\\"image__user css-ozpmrz\\">by <!-- -->12019</div></div></li></ul></div></div><style data-emotion-css=\\"1xjzuas\\">.css-1xjzuas{text-align:center;box-sizing:border-box;}@media (max-width:375px){.css-1xjzuas{position:absolute;top:0;left:0;width:100%;height:100%;background-color:transparent;padding:0 14px;pointer-events:none;}}</style><style data-emotion-css=\\"6ro6yz\\">.css-6ro6yz{background-color:#E4EBE0;padding:16px;text-align:center;box-sizing:border-box;}@media (max-width:375px){.css-6ro6yz{padding-bottom:32px;}}@media (max-width:375px){.css-6ro6yz{position:absolute;top:0;left:0;width:100%;height:100%;background-color:transparent;padding:0 14px;pointer-events:none;}}</style><div class=\\"carousel__controls css-6ro6yz\\"><style data-emotion-css=\\"bx1xrk\\">.css-bx1xrk position:\\"relative\\",.css-bx1xrk width:\\"100%\\",.css-bx1xrk height:\\"100%\\",.css-bx1xrk pointerEvents:\\"none\\" @media (max-width:375px){width:\\"100%\\", height:\\"100%\\",;}</style><div class=\\"carousel__controls-buttons css-bx1xrk\\"><style data-emotion-css=\\"13brkk7\\">.css-13brkk7{margin-right:6px;}</style><style data-emotion-css=\\"tuuwt7\\">.css-tuuwt7{pointer-events:all;display:inline-block;background-color:#035288;color:#fff;border:none;padding:6px 12px;cursor:pointer;-webkit-transition:all 0.2s ease-in-out;transition:all 0.2s ease-in-out;outline:none;border-radius:5px 0 0 5px;margin-right:6px;}.css-tuuwt7:hover{background-color:#fff;color:#035288;}@media (max-width:375px){.css-tuuwt7{background-color:rgba(255,255,255,0.5);height:120px;width:120px;border-radius:50%;position:absolute;top:38%;}}@media (max-width:375px){.css-tuuwt7{left:-60px;}}</style><button class=\\"carousel__prev css-tuuwt7\\" tabindex=\\"0\\"><style data-emotion-css=\\"1o7umnt\\">.css-1o7umnt{font-family:\\"Ropa Sans\\",sans-serif;font-size:18px;}@media (max-width:375px){.css-1o7umnt{display:none;}}</style><span class=\\"button__text css-1o7umnt\\">Prev</span><style data-emotion-css=\\"1h2v11t\\">.css-1h2v11t{display:none;text-align:left;-webkit-transform:scaleX(-1);-ms-transform:scaleX(-1);transform:scaleX(-1);}.css-1h2v11t svg{display:inline-block;}.css-1h2v11t svg path{fill:#035288;}@media (max-width:375px){.css-1h2v11t{display:block;padding-left:12px;}}</style><div class=\\"button__icon css-1h2v11t\\">svg-icon-stub</div></button><style data-emotion-css=\\"2hvxwp\\">.css-2hvxwp{pointer-events:all;display:inline-block;background-color:#035288;color:#fff;border:none;padding:6px 12px;cursor:pointer;-webkit-transition:all 0.2s ease-in-out;transition:all 0.2s ease-in-out;outline:none;border-radius:0 5px 5px 0;}.css-2hvxwp:hover{background-color:#fff;color:#035288;}@media (max-width:375px){.css-2hvxwp{background-color:rgba(255,255,255,0.5);height:120px;width:120px;border-radius:50%;position:absolute;top:38%;}}@media (max-width:375px){.css-2hvxwp{right:-60px;}}</style><button class=\\"carousel__next css-2hvxwp\\" tabindex=\\"0\\"><span class=\\"button__text css-1o7umnt\\">Next</span><style data-emotion-css=\\"1mk1qkb\\">.css-1mk1qkb{display:none;text-align:left;}.css-1mk1qkb svg{display:inline-block;}.css-1mk1qkb svg path{fill:#035288;}@media (max-width:375px){.css-1mk1qkb{display:block;padding-left:12px;}}</style><div class=\\"button__icon css-1mk1qkb\\">svg-icon-stub</div></button></div></div></div>"`;
